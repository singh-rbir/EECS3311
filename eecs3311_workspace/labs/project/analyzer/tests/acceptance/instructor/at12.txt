-- A simple bank account example 

generate_java_code
add_class("Account")
generate_java_code
add_attribute("Account", "balance", "INTEGER")
generate_java_code
add_command("Account", "set_balance", <<["new_balance", "INTEGER"]>>)
generate_java_code
add_query("Account", "get_balance", <<>>, "INTEGER")
generate_java_code

add_assignment_instruction("Account", "set_balance", "balance")
generate_java_code
add_call_chain(<<"new_balance">>)
generate_java_code

add_assignment_instruction("Account", "get_balance", "Result")
generate_java_code
add_call_chain(<<"balance">>)
generate_java_code
