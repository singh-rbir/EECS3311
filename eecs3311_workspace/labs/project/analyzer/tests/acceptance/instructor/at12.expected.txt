  Status: OK.
  Number of classes being specified: 0
->generate_java_code

->add_class("Account")
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->generate_java_code
  class Account {
  }
->add_attribute("Account","balance","INTEGER")
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 0
      Number of commands: 0
->generate_java_code
  class Account {
    int balance;
  }
->add_command("Account","set_balance",<<["new_balance", "INTEGER"]>>)
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_balance(INTEGER)
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
    }
  }
->add_query("Account","get_balance",<<>>,"INTEGER")
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 1
        + get_balance: INTEGER
      Number of commands: 1
        + set_balance(INTEGER)
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
    }
    int get_balance() {
      int Result = 0;
      return Result;
    }
  }
->add_assignment_instruction("Account","set_balance","balance")
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 1
        + get_balance: INTEGER
      Number of commands: 1
        + set_balance(INTEGER)
  Routine currently being implemented: {Account}.set_balance
  Assignment being specified: balance := ?
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
      balance = ?;
    }
    int get_balance() {
      int Result = 0;
      return Result;
    }
  }
->add_call_chain(<<"new_balance">>)
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 1
        + get_balance: INTEGER
      Number of commands: 1
        + set_balance(INTEGER)
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
      balance = new_balance;
    }
    int get_balance() {
      int Result = 0;
      return Result;
    }
  }
->add_assignment_instruction("Account","get_balance","Result")
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 1
        + get_balance: INTEGER
      Number of commands: 1
        + set_balance(INTEGER)
  Routine currently being implemented: {Account}.get_balance
  Assignment being specified: Result := ?
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
      balance = new_balance;
    }
    int get_balance() {
      int Result = 0;
      Result = ?;
      return Result;
    }
  }
->add_call_chain(<<"balance">>)
  Status: OK.
  Number of classes being specified: 1
    Account
      Number of attributes: 1
        + balance: INTEGER
      Number of queries: 1
        + get_balance: INTEGER
      Number of commands: 1
        + set_balance(INTEGER)
->generate_java_code
  class Account {
    int balance;
    void set_balance(int new_balance) {
      balance = new_balance;
    }
    int get_balance() {
      int Result = 0;
      Result = balance;
      return Result;
    }
  }
